state/migration_export_test.go

```diff
@@ -1125,6 +1125,10 @@ func (s *MigrationExportSuite) TestStoragePools(c *gc.C) {
 }
 
 func (s *MigrationExportSuite) TestPayloads(c *gc.C) {
+	s.PatchValue(state.ValidPayloadForUnit, func(unit *state.Unit, c, t string) error {
```

Comment:
This is not going to b easy because to use the charm here would also mean that I'd need valid unit assigned to a machine with instance data and agent tools, an application, etc... I have gone through this dance using Factories and all but then the test will no longer be a unit test but a full stack one and will change enough to obscure the actual functionality under test "can we export payload data"